src/Gold.Redis/Gold.Redis.LowLevelClient/Communication/RedisSocketAuthenticator.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Gold.Redis.Common.Interfaces.Communication;
+using System;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+
+namespace Gold.Redis.LowLevelClient.Communication
+{
+    public class RedisSocketAuthenticator : IRedisAuthenticator
+    {
+        private readonly ISocketCommandExecutor _socketCommandExecutor;
+        public RedisSocketAuthenticator(ISocketCommandExecutor socketCommandExecutor)
+        {
+            _socketCommandExecutor = socketCommandExecutor;
+        }
+        public async Task<bool> TryAuthenticate(Socket connectionSocket, string password)

```

Comment:
i dont really like this class, why cant it use RedisCommandHandler instead of fully touching the socket?
also cant it be just a normal command?